### PR TITLE
🍱 Added gitmoji data file

### PR DESCRIPTION
**Pull request by tusqasi**
Adding gitmoji symbols from https://gitmoji.dev/

### CODE REVIEW

**Comment by fdw** [comment]
This looks awesome, thank you!

Have you created the file by hand or is there some automation? I'm just wondering how to keep this file up to date...

**Comment by tusqasi** [comment]
I downloaded the json file which has all the descriptions and emojis from gitmoji repo, then made a python script to read it and save to a file.

**Comment by fdw** [comment]
Do you think we could add your script as another extractor in `src/extractors`? If you need more details, just ask :)

**Comment by fdw** [comment]
Hi @tusqasi, would you mind if I try to write an extractor for gitmoji? Or do you still want to?

**Comment by tusqasi** [comment]
I am sorry, I didn't read the last comment. I have started working on an extractor right now.   

I am using a file from a github repo for getting the gitmojis, would that raise any issues you know of? Other than that, no doubts now.

**Comment by fdw** [comment]
As long as the license allows that, I don't see any problem 🙂
Thank you!

**Comment by tusqasi** [comment]
Hope that is what you expect out of an extractor. Let me know of any changes.

File: src/extractors/gitmojiextractor.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+"""
```
Comment (fdw) [inline src/extractors/gitmojiextractor.py:1]: I don't think this comment adds anything. What do you say?

File: src/extractors/gitmojiextractor.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Copied and changed from nerdextractor.py in https://github.com/fdw/rofimoji
+"""
+
+
+import html
+from pathlib import Path
+import json
```
Comment (fdw) [inline src/extractors/gitmojiextractor.py:8]: `json` is never used, so we can probably delete it.

File: src/extractors/gitmojiextractor.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            'emoji': '🎨',
+            'entity': '&#x1f3a8;',
+            'code': ':art:',
+            'description': 'Improve structure / format of the code.',
+            'name': 'art',
+            'semver': None
+        }
+        """
+        for char in response.json()["gitmojis"]:
+            icon = ord(char["emoji"][0])
```
Comment (fdw) [inline src/extractors/gitmojiextractor.py:37]: Are we sure there is only ever one character?

File: src/extractors/gitmojiextractor.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            name = char["description"]
+            self.icons.append(Character(icon, name))
+
+    def write_to_file(self: 'GitmojiExtractor'):
+        if len(self.icons) == 0:
```
Comment (fdw) [inline src/extractors/gitmojiextractor.py:42]: This is just a shortcut, right?

i'm not sure if it's easier to understand with the `if` than without... what do you say?

**Review by fdw — COMMENTED** [review]
Looks good, thank you very much 🙂

Please have a look at my comments, but there's nothing major.

File: src/extractors/gitmojiextractor.py (hunk 2 above)
Comment (tusqasi) [reply src/extractors/gitmojiextractor.py:8]: Makes sense

File: src/extractors/gitmojiextractor.py (hunk 3 above)
Comment (tusqasi) [reply src/extractors/gitmojiextractor.py:37]: I am using only the first character of the string. No issues with it so far.

File: src/extractors/gitmojiextractor.py (hunk 4 above)
Comment (tusqasi) [reply src/extractors/gitmojiextractor.py:42]: Now this isn't my code, I don't know why this check was added. Best to keep the check there, I think.

File: src/extractors/gitmojiextractor.py (hunk 1 above)
Comment (tusqasi) [reply src/extractors/gitmojiextractor.py]: On second thought, that doesn't really add anything.

File: src/extractors/gitmojiextractor.py (hunk 4 above)
Comment (fdw) [reply src/extractors/gitmojiextractor.py:36]: You mean you copied this from `nerdextractor.py`, right?

In any case, we can leave this, but I don't think it adds anything. If `self.icons` is empty, the `for` loop doesn't execute and thus nothing is written anyway.

Comment (tusqasi) [reply src/extractors/gitmojiextractor.py:36]: Yeah, makes sense. I'll just remove it.

**Comment by tusqasi** [comment]
Let me know of any more changes.

**Comment by fdw** [comment]
Awesome, thank you very much! 🙂

**Comment by tusqasi** [comment]
:+1:

**Pull request merged**